scrape_linkedin/Profile.py

```diff
@@ -150,6 +151,94 @@ def interests(self):
             i, '.pv-entity__summary-title'), interests)
         return list(interests)
 
+    @property
+    def recommendations(self):
+        recs = {
+            'received': [],
+            'given': [],
+        }
+        rec_block = one_or_default(self.soup, '.recommendations-inlining')
+        if rec_block:
+            div_blocks = rec_block.find_all('div', recursive=False)
+            if div_blocks:
+                rec_tabs = div_blocks[-1]
+                if rec_tabs:
+                    buttons_and_rec_blocks = rec_tabs.find_all('div', recursive=False)
+                    if buttons_and_rec_blocks:
+                        buttons = all_or_default(buttons_and_rec_blocks[0], 'button')
+                        if buttons:
+                            expr = re.compile(r'((?<=in\/).+(?=\/)|(?<=in\/).+)')  # re to get li id
+                            date_expr = re.compile(r'\w+ \d{2}, \d{4}, ')  # re to get date of recommendation
+                            for idx, b in enumerate(buttons):
+                                for rec_type in recs.keys():
+                                    if rec_type in str(b.contents).lower():
+                                        recs_category = buttons_and_rec_blocks[idx + 1]  # +1 --> skip buttons block
+                                        for rec in recs_category.find_all('li', {'class': 'pv-recommendation-entity'}):
+                                            rec_dict = {
+                                                'text': None,
+                                                'date': None,
+                                                'relationship': None,
+                                                'recommender': {
+                                                    'firstName': None,
+                                                    'lastName': None,
```

Comment:
I think having firstName and lastName is unnecessary, and has some edge cases with people from certain countries. I would suggest just a single 'name', which also matches up with what is recorded for a profile. If individuals doing scrapes want to post-process full names into first/last names based on their own localized assumptions, they can do that.

Reply:
Agree